pykeybasebot/kvstore_client.py

```diff
@@ -0,0 +1,115 @@
+import json
+from typing import Any, Dict, Union
+
+from .errors import DeleteNonExistentError, RevisionError
+from .types import keybase1
+
+
+class KVStoreClient:
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def put(
+        self,
+        team: str,
+        namespace: str,
+        entryKey: str,
+        entryValue: str,
+        revision: Union[int, None] = None,
+    ) -> keybase1.KVPutResult:
+        await self.bot.ensure_initialized()
+        args: Dict[str, Any] = {
+            "method": "put",
+            "params": {
+                "options": {
+                    "team": team,
+                    "namespace": namespace,
+                    "entryKey": entryKey,
+                    "entryValue": entryValue,
+                }
+            },
+        }
+        if revision:
+            args["params"]["options"]["revision"] = revision
+        try:
+            res = await self.execute(args)
+            return keybase1.KVPutResult.from_dict(res)
+        except Exception as e:
+            if e.args[0]["code"] == 2760:
+                raise RevisionError(e.args[0]["message"])
+            else:
+                raise e
+
+    async def delete(
+        self,
+        team: str,
+        namespace: str,
+        entryKey: str,
+        revision: Union[int, None] = None,
+    ) -> keybase1.KVDeleteEntryResult:
+        await self.bot.ensure_initialized()
+        args: Dict[str, Any] = {
+            "method": "del",
+            "params": {
+                "options": {"team": team, "namespace": namespace, "entryKey": entryKey}
+            },
+        }
+        if revision:
+            args["params"]["options"]["revision"] = revision
+        try:
+            res = await self.execute(args)
+            return keybase1.KVDeleteEntryResult.from_dict(res)
+        except Exception as e:
+            if e.args[0]["code"] == 2762:
+                raise DeleteNonExistentError(e.args[0]["message"])
+            elif e.args[0]["code"] == 2760:
```

Comment:
this makes me think perhaps this logic should be in a function over in errors that knows how to distinguish between different high-level errors based on the service response. and then that file can encapsulate all the error handling logic and constants and specific knowledge (e.g. 2762 --> DeleteNonExistentError) and over here all we need to do is delegate over to the error handler over there. 
not a big deal. it's very unlikely we'll have many more of these.